service_crategen/src/commands/generate/codegen/mod.rs

```diff
@@ -313,31 +319,20 @@ fn generate_types<P>(writer: &mut FileWriter, service: &Service, protocol_genera
         if is_streaming_shape(service, name) {
             // Add a second type for streaming blobs, which are the only streaming type we can have
             writeln!(writer,
-                     "pub struct {streaming_name}(Box<Read>);
+                     "pub struct {streaming_name}(Box<FuturesStream<Item=Vec<u8>, Error=HttpDispatchError> + Send>);
 
                      impl fmt::Debug for {streaming_name} {{
                          fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {{
                              write!(f, \"<{name}: streaming content>\")
                          }}
                      }}
 
-                     impl ::std::ops::Deref for {streaming_name} {{
-                         type Target = Box<Read>;
-
-                         fn deref(&self) -> &Box<Read> {{
-                             &self.0
-                         }}
-                     }}
-
-                     impl ::std::ops::DerefMut for {streaming_name} {{
-                         fn deref_mut(&mut self) -> &mut Box<Read> {{
-                             &mut self.0
-                         }}
-                     }}
+                     impl FuturesStream for {streaming_name} {{
+                         type Item = Vec<u8>;
+                         type Error = HttpDispatchError;
 
-                     impl ::std::io::Read for {streaming_name} {{
-                         fn read(&mut self, buf: &mut [u8]) -> ::std::io::Result<usize> {{
-                             self.0.read(buf)
+                         fn poll(&mut self) -> Poll<Option<Self::Item>, Self::Error> {{
+                             self.0.poll()
```

Comment:
Again I'd prefer the full struct here. Maybe:

```
"pub struct {streaming_name} {
  {streaming_name}_stream: Box<FutureStream<Item=Vec<u8>, Error=HttpDispatchError> + send>);
}
```

Reply:
Sure!

- [ ] change tuple struct to proper struct with named fields